cmd/erasure-multipart.go

```diff
@@ -568,6 +580,12 @@ func (er erasureObjects) ListObjectParts(ctx context.Context, bucket, object, up
 		return result, toObjectErr(err, bucket, object, uploadID)
 	}
 
+	uploadIDLock := er.NewNSLock(ctx, bucket, pathJoin(object, uploadID))
```

Comment:
Since this is after `checkUploadIDExists` it could have been deleted before the lock has been acquired AFAICT.

Maybe it will be caught at `er.getUploadIDDir` but then the check above doesn't do much more than early rejection, but probably is just superfluous.